xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -348,60 +349,88 @@ bool CGUIWindowPVRBase::PlayFile(CFileItem *item, bool bPlayMinimized /* = false
   return true;
 }
 
-bool CGUIWindowPVRBase::StartRecordFile(const CFileItem &item)
+bool CGUIWindowPVRBase::ShowTimerSettings(CFileItem *item)
 {
-  if (!item.HasEPGInfoTag())
+  if (!item->IsPVRTimer())
     return false;
 
-  const CEpgInfoTagPtr tag = item.GetEPGInfoTag();
+  CGUIDialogPVRTimerSettings* pDlgInfo = (CGUIDialogPVRTimerSettings*)g_windowManager.GetWindow(WINDOW_DIALOG_PVR_TIMER_SETTING);
+
+  if (!pDlgInfo)
+    return false;
+
+  pDlgInfo->SetTimer(item);
+  pDlgInfo->DoModal();
+
+  return pDlgInfo->IsConfirmed();
+}
+
+bool CGUIWindowPVRBase::StartRecordFile(CFileItem *item, bool bAdvanced)
+{
+  if (!item->HasEPGInfoTag())
+    return false;
+
+  const CEpgInfoTagPtr tag = item->GetEPGInfoTag();
   CPVRChannelPtr channel = tag->ChannelTag();
 
   if (!channel || !g_PVRManager.CheckParentalLock(channel))
     return false;
 
-  CFileItemPtr timer = g_PVRTimers->GetTimerForEpgTag(&item);
+  CFileItemPtr timer = g_PVRTimers->GetTimerForEpgTag(item);
   if (timer && timer->HasPVRTimerInfoTag())
   {
     CGUIDialogOK::ShowAndGetInput(19033, 19034);
     return false;
   }
 
-  // ask for confirmation before starting a timer
-  CGUIDialogYesNo* pDialog = (CGUIDialogYesNo*)g_windowManager.GetWindow(WINDOW_DIALOG_YES_NO);
-  if (!pDialog)
-    return false;
-  pDialog->SetHeading(264);
-  pDialog->SetLine(0, tag->PVRChannelName());
-  pDialog->SetLine(1, "");
-  pDialog->SetLine(2, tag->Title());
-  pDialog->DoModal();
+  bool bReturn(false);
 
-  if (!pDialog->IsConfirmed())
-    return false;
+  if (bAdvanced)
+  {
+    CPVRTimerInfoTagPtr newTimer = CPVRTimerInfoTag::CreateFromEpg(tag, true);
+    if (newTimer)
+    {
+      CFileItem *newItem = new CFileItem(newTimer);
 
-  CPVRTimerInfoTagPtr newTimer = CPVRTimerInfoTag::CreateFromEpg(tag);
-  bool bReturn(false);
-  if (newTimer)
+      if (ShowTimerSettings(newItem))
+        bReturn = g_PVRTimers->AddTimer(newItem->GetPVRTimerInfoTag());
+
+      delete newItem;
+    }
+  }
+  else
   {
-    bReturn = g_PVRTimers->AddTimer(newTimer);
+    // ask for confirmation before starting a timer
+    if (!CGUIDialogYesNo::ShowAndGetInput(
+          264 /* "Stop Rec." */, tag->PVRChannelName(), "", tag->Title()))
+      return false;
+
+    CPVRTimerInfoTagPtr newTimer = CPVRTimerInfoTag::CreateFromEpg(tag);
+
+    if (newTimer)
+      bReturn = g_PVRTimers->AddTimer(newTimer);
   }
   return bReturn;
 }
 
-bool CGUIWindowPVRBase::StopRecordFile(const CFileItem &item)
+bool CGUIWindowPVRBase::StopRecordFile(CFileItem *item)
```

Comment:
To make it consitent with all the other methods in this class. Alternative would have been to change all signatures to const CFileItem&. Decided to change just one or two.